modules/azure/compute.go

```diff
@@ -0,0 +1,102 @@
+package azure
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute"
+	"github.com/stretchr/testify/require"
+)
+
+// getVirtualMachineClient is a helper function that will setup an Azure Virtual Machine client on your behalf
+func getVirtualMachineClient(subscriptionID string) (compute.VirtualMachinesClient, error) {
+	// Validate Azure subscription ID
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return compute.NewVirtualMachinesClient(""), err
+	}
+
+	// Create a VM client
+	vmClient := compute.NewVirtualMachinesClient(subscriptionID)
+
+	// Create an authorizer
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return compute.NewVirtualMachinesClient(""), err
+	}
+
+	// Attach authorizer to the client
+	vmClient.Authorizer = *authorizer
+
+	return vmClient, nil
+}
+
+// GetSizeOfVirtualMachine gets the size type of the given Azure Virtual Machine
+func GetSizeOfVirtualMachine(t *testing.T, vmName string, resGroupName string, subscriptionID string) string {
+	size, err := GetSizeOfVirtualMachineE(t, vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return size
+}
+
+// GetSizeOfVirtualMachineE gets the size type of the given Azure Virtual Machine
+func GetSizeOfVirtualMachineE(t *testing.T, vmName string, resGroupName string, subscriptionID string) (string, error) {
+	// Validate resource group name and subscription ID
+	resGroupName, err := getTargetAzureResourceGroupName(resGroupName)
+	if err != nil {
+		return "", err
+	}
+
+	// Create a VM client
+	vmClient, err := getVirtualMachineClient(subscriptionID)
+	if err != nil {
+		return "", err
+	}
+
+	// Get the details of the target virtual machine
+	vm, err := vmClient.Get(context.Background(), resGroupName, vmName, compute.InstanceView)
+	if err != nil {
+		return "", err
+	}
+
+	return string(vm.VirtualMachineProperties.HardwareProfile.VMSize), nil
```

Comment:
What type is `VMSize`?

Reply:
`VMSize` is a custom type called `VirtualMachineSizeTypes`, which enumerates different values.  Documented here:

https://godoc.org/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute#VirtualMachineSizeTypes

I didn't want to have people have to assert types directly out of the Azure SDK to verify results.  If you see a cleaner way, happy to discuss, totally possible I'm missing something obvious here.

Reply:
I suspect if you're testing Azure, you're using the Azure SDK anyway, so using an enum from there would be more convenient than stringly-typed programming.